core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationEditorDirective.js

```diff
@@ -0,0 +1,149 @@
+                backdrop: true,
+                resolve: {
+                  considerTranslation: function() {
+                    return false;
+                  }
+                },
+                controller: 'MarkAllAudioAsNeedingUpdateController'
+              }).result.then(function() {
+                contentIdsToAudioTranslations.markAllAudioAsNeedingUpdate(
+                  contentId);
+                ExplorationStatesService.saveContentIdsToAudioTranslations(
+                  stateName, contentIdsToAudioTranslations);
+              });
+            }
+          };
+          var contentId = null;
+          var langaugeCode = null;
+          $scope.activeWrittenTranslation = null;
+          $scope.translationEditorIsOpen = false;
+          $scope.isTranslatable = EditabilityService.isTranslatable;
+          $scope.$on('activeContentIdChanged', function() {
+            initEditor();
+          });
+          $scope.$on('activeLanguageChanged', function() {
+            initEditor();
+          });
+
+          var initEditor = function() {
+            $scope.activeWrittenTranslation = null;
+            $scope.translationEditorIsOpen = false;
+            contentId = (
+              TranslationTabActiveContentIdService.getActiveContentId());
+            langaugeCode = TranslationLanguageService.getActiveLanguageCode();
+            if (StateWrittenTranslationsService.displayed.hasWrittenTranslation(
+              contentId, langaugeCode)) {
+              $scope.activeWrittenTranslation = (
+                StateWrittenTranslationsService.displayed
+                  .getWrittenTranslation(contentId, langaugeCode));
+            }
+          };
+          initEditor();
+          var saveTranslation = function() {
+            var oldTranslation = '';
+            var newTranslation = '';
+            contentId = (
+              TranslationTabActiveContentIdService.getActiveContentId());
+            langaugeCode = TranslationLanguageService.getActiveLanguageCode();
+            if (StateWrittenTranslationsService
+              .savedMemento.hasWrittenTranslation(contentId, langaugeCode)) {
+              oldTranslation = (StateWrittenTranslationsService
+                .savedMemento.getWrittenTranslation(contentId, langaugeCode));
+            }
+            newTranslation = (StateWrittenTranslationsService
+              .displayed.getWrittenTranslation(contentId, langaugeCode));
+            if (oldTranslation !== newTranslation) {
+              var stateName = StateEditorService.getActiveStateName();
+              showMarkAllAudioAsNeedingUpdateModalIfRequired(contentId);
+              StateWrittenTranslationsService.saveDisplayedValue();
+              ExplorationStatesService.saveWrittenTranslations(
+                stateName, StateWrittenTranslationsService.savedMemento);
+            }
+            $scope.translationEditorIsOpen = false;
```

Comment:
Why though? I need a general clarification on this variable. translationEditor refers to the voiceover or the written translations? So this above function saves a written translation, and once saved why do we close the translation editor?

Reply:
Oh I see, so audio isn't a translation... I see, now I kind of understand! Thanks!